Data/ByteString/Internal.hs

```diff
@@ -270,6 +274,24 @@ instance Data ByteString where
   gunfold _ _    = error "Data.ByteString.ByteString.gunfold"
   dataTypeOf _   = mkNoRepType "Data.ByteString.ByteString"
 
+-- | @since 0.11.2.0
+instance TH.Lift ByteString where
+#if MIN_VERSION_template_haskell(2,16,0)
+  lift (BS ptr len) = [| unsafePackLenLiteral |]
+    `TH.appE` TH.litE (TH.integerL (fromIntegral len))
+    `TH.appE` TH.litE (TH.BytesPrimL $ TH.Bytes ptr 0 (fromIntegral len))
+#else
+  lift bs@(BS _ len) = [| unsafePackLenLiteral |]
+    `TH.appE` TH.litE (TH.integerL (fromIntegral len))
```

Comment:
Could we possibly have less duplication? It seems that only the last line is conditional on `template-haskell` version. Same for `SBS`.

Reply:
it's not the same,

```haskell
lift (BS ptr len) 
```

vs

```
  lift bs@(BS _ len)
```

Disable unused variable warning and I will simpify.